src/Compilers/CSharp/Portable/Lowering/AsyncRewriter/AsyncMethodToStateMachineRewriter.cs

```diff
@@ -188,6 +199,16 @@ internal void GenerateMoveNext(BoundStatement body, MethodSymbol moveNextMethod)
                             ? ImmutableArray.Create<BoundExpression>(F.Local(_exprRetValue))
                             : ImmutableArray<BoundExpression>.Empty)));
 
+            if (_asyncIteratorInfo != null)
+            {
+                bodyBuilder.Add(
+                    // if (this.promiseIsActive)
+                    // {
+                    //    this.promiseOfValueOrEnd.SetResult(false);
+                    // }
+                    GenerateSetResultOnPromiseIfActive(false));
```

Comment:
How does this interact with the ordering of notifying the sync context above? Why do we want to ping the sync context first and set the promise second?